integrationtests/foldersync/client.go

```diff
@@ -0,0 +1,346 @@
+package foldersync
```

Comment:
All the folder sync code is mostly ported (cleaned up a bit along the way) or copy/pasted/updated from the old folder sync repo.

Reply:
I put it in this integration tests directory. We can move other things that are actually integration tests in here as well.